src/qibo/gates/gates.py

```diff
@@ -2469,6 +2469,76 @@ def __init__(self, q0, q1, q2, theta, trainable=True):
         self.init_kwargs = {"theta": theta, "trainable": trainable}
 
 
+class GeneralizedRBS(ParametrizedGate):
+    """The generalized Reconfigurable Beam Splitter gate.
+
+    Corresponds to the following matrix:
+
+    .. math::
+
+        \begin{pmatrix}
+            I &           &  &             &  \\
+              & e^{i\\phi}\\cos\theta &      & e^{-i\\phi}\\sin\theta   &  \\
+              &  & I'       &    &  \\
```

Comment:
Why do you need an explicit identity here?

This gate is always acting on just two qubits, can't you just define the matrix on the subspace of the two qubits involved?
I would keep this similar to the swap gate, if possible...

Reply:
This is not a two-qubit matrix. I forgot to include that in the docstring.